internal/nginx/modules/httpmatches.js

```diff
@@ -0,0 +1,83 @@
+export default { redirect };
+
+const matchesVariable = 'http_matches';
+
+// FIXME(osborn): Need to add special handling for repeated headers.
+// Should follow guidance from https://www.rfc-editor.org/rfc/rfc7230.html#section-3.2.2.
+function headersMatch(r, headers) {
+  for (let i = 0; i < headers.length; i++) {
+    const h = headers[i];
+    const kv = h.split(':');
+    if (kv.length !== 2) {
+      r.error('invalid header match: ' + h);
+      return false;
+    }
+    let val = r.headersIn[kv[0]];
+    if (!val || val.toLowerCase() !== kv[1]) {
+      r.warn('header: ' + h + ' does not exist in request');
+      return false;
+    }
+  }
+
+  return true;
+}
+
+function paramsMatch(r, params) {
+  for (let i = 0; i < params.length; i++) {
+    let p = params[i];
+    // get index of first = in string
+    const idx = p.indexOf('=');
+    // divide string into key value using the index
+    let kv = [p.slice(0, idx), p.slice(idx + 1)];
+    if (kv.length !== 2) {
```

Comment:
I don't think this is possible based on the line above.

Reply:
Thanks, changed.